design-proposals/klip-63-pause-resume-persistent-queries.md

```diff
@@ -0,0 +1,62 @@
+# KLIP-63: PAUSE and RESUME for persistent queries 
+
+**Author**: Jim Hughes (jnh5y) | 
+**Release Target**: Unknown | 
+**Status**: In Discussion | 
+**Discussion**: 
+
+**tl;dr:** Adds the ability to pause the processing of a persistent query.
+           
+## Motivation and background
+
+Users may wish to pause a query for a number of reasons and then resume processing.  Presently, they would have to terminate a query and recreate it.  This approach means that any state would have to be restored; the goal of this work is to add the capability to pause processing temporarily while allowing Kafka consumers and state stores to be retained.
```

Comment:
will be good to enumerate a few reasons users might want to pause a persistent query

Reply:
Sure, I updated some words here.  Feel free to suggest other use cases!

Reply:
new text looks good. In addition to those, is improved responsiveness on the UI also a use case at all for streaming designer? If there are multiple persistent queries running concurrently, does pausing a few ensure others may make faster progress?

Reply:
I don't know if pausing queries will help with the UI for streaming designer; that may depend on implementation details of that product.  I was trying to keep the motivation general.

In some situations, I could imagine pausing a portion of the queries would allow the remainder to run faster.  I was trying to suggest that with the new text "... or to manage computational resources."

Should I be more concrete?

Also, just as a note, paused queries will still use memory and CPU for the Kafka consumers.  Given that, I don't think folks will be using PAUSE and RESUME to address lag in a data pipeline.  (It'd like be better to scale the cluster or modify the flow to achieve the desired latency.)